python/_tskitmodule.c

```diff
@@ -6288,6 +6288,39 @@ TableCollection_subset(TableCollection *self, PyObject *args)
     return ret;
 }
 
+static PyObject *
+TableCollection_union(TableCollection *self, PyObject *args)
+{
+    int err;
+    TableCollection *other = NULL;
+    PyObject *ret = NULL;
+    PyObject *nodes = NULL;
+    PyArrayObject *nodes_array = NULL;
+    npy_intp *shape;
+    tsk_size_t num_nodes;
+
+    if (!PyArg_ParseTuple(args, "O!O", &TableCollectionType, &other, &nodes)) {
```

Comment:
@petrelharp can you help me with the parsing of the arguments? I'm doing this, but I don't think it's right.

Reply:
Hm, that is not obviously wrong to me, based on comparing to e.g. `simplify()` and `table_collection-equals()` - does it work?

Reply:
I had to do a forward declaration of `TableCollectionType`. Not entirely sure why that is, tho. But it worked!

Reply:
I mean, I understand variables have to be declared in C, but not sure why I have to do that outside the function for this struct.

Reply:
It's because it's defined later on in this file. I see there's another forward declaration before TableCollection_equals (probably where you got this?) - that one could be deleted now that you've got this in here. But, style-wise I don't know why the definition comes later on, or if it's good practice to leave both forward declarations in...

Reply:
Usually we try to arrange the file so that types are defined before they are used so we don't have to bother maintaining two copies of declarations. Sometimes you get chicken and egg situations though, and a forward declaration is needed. There's no particular reason for ordering of stuff in the file otherwise (other than keeping all the stuff for a particular class adjacent).